src/Common/tests/System/PlatformDetection.cs

```diff
@@ -23,6 +23,32 @@ public static class PlatformDetection
 
         public static int WindowsVersion { get; } = GetWindowsVersion();
 
+        private static Lazy<bool> m_isWindowsSubsystemForLinux = new Lazy<bool>(GetIsWindowsSubsystemForLinux);
+
+        public static bool IsWindowsSubsystemForLinux => m_isWindowsSubsystemForLinux.Value;
+        public static bool IsNotWindowsSubsystemForLinux => !IsWindowsSubsystemForLinux;
+
+        private static bool GetIsWindowsSubsystemForLinux()
+        {
+            // https://github.com/Microsoft/BashOnWindows/issues/423#issuecomment-221627364
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                const string versionFile = "/proc/version";
+                if (File.Exists(versionFile))
+                {
+                    var s = File.ReadAllText(versionFile);
+
+                    if (s.Contains("Microsoft") || s.Contains("WSL"))
```

Comment:
What is the chance that some Linux distribution will contain `WSL` as a substring in its `/proc/version`?